pdns/pdns_recursor.cc

```diff
@@ -4109,35 +4110,40 @@ static FDMultiplexer* getMultiplexer()
 }
 
 
-static string* doReloadLuaScript()
+static RecursorControlChannel::Answer* doReloadLuaScript()
 {
   string fname= ::arg()["lua-dns-script"];
   try {
     if(fname.empty()) {
       t_pdl.reset();
       g_log<<Logger::Info<<t_id<<" Unloaded current lua script"<<endl;
-      return new string("unloaded\n");
+      return new RecursorControlChannel::Answer{0, string("unloaded\n")};
     }
     else {
       t_pdl = std::make_shared<RecursorLua4>();
-      t_pdl->loadFile(fname);
+      int err = t_pdl->loadFile(fname);
```

Comment:
Not related to this PR but adding the same kind of check on the initial loading of the file would fix https://github.com/PowerDNS/pdns/issues/10079